distributed/dashboard/scheduler.py

```diff
@@ -292,6 +293,20 @@ def collect(self):
             sum, (tp.states for tp in self.server.task_prefixes.values())
         )
 
+        suspicious_tasks = CounterMetricFamily(
+            "dask_scheduler_tasks_suspicious",
+            "Total number of times a task has been marked suspicious",
+            labels=["task_name"],
+        )
+
+        suspicious_by_prefix = Counter()
+        for task in self.server.tasks.values():
```

Comment:
In general we try to avoid looping over all tasks where possible. This can noticeably slow things down when there are millions of tasks. Do you have any thoughts on how we might improve things here?

Reply:
The only thing I can think of right now would be keeping the `suspicious_by_prefix` counter at the scheduler similar to the unrunnable tasks and increasing this counter every time a task is marked as suspicious.

Reply:
That seems reasonable to me.

Reply:
Instead of increasing the counter similar to `unrunnable` we might want to include this in the `TaskGroup`/`TaskPrefix` containers. Should be similar in complexity but the group/prefix containers seem to be the "future" way of aggregating task state

Reply:
As my counter is already called `suspicious_by_prefix` putting it into the `TaskPrefix` seems very sensible to me. Thanks @fjetter

Reply:
Are the TaskPrefixes cleared after a graph is calculated?

Reply:
`TaskGroup`s are deleted up when all the tasks in the group are in the `'forgotten'` state, but `TaskPrefixes` are not removed

Reply:
So we would have to use `TaskGroups` in order to not potentially have a wrong counter for the tasks, but with the current state of `TaskGroups` for example for `delayed` objects there seems to be one TaskGroup per task. This would not seem feasible to me and I would stay with the implementation at the Scheduler here for now. Are there any comments on this? Is the thing with the TaskGroups for delayed actually correct?